falcon/routing/compiled.py

```diff
@@ -362,4 +358,4 @@ def conflicts_with(self, segment):
                 #   /foo/all
                 return True
```

Comment:
This is the opposite case, and shouldn't be changed.  A variable followed by a static route would be unreachable.